src/Symfony/Component/HttpFoundation/Session/Storage/Proxy/SessionHandlerProxy.php

```diff
@@ -70,22 +76,25 @@ public function write($sessionId, $data)
     /**
      * @return bool
      */
+    #[ReturnTypeWillChange]
     public function destroy($sessionId)
     {
         return (bool) $this->handler->destroy($sessionId);
     }
 
     /**
-     * @return bool
+     * @return int|false
      */
+    #[ReturnTypeWillChange]
     public function gc($maxlifetime)
     {
-        return (bool) $this->handler->gc($maxlifetime);
+        return (int) $this->handler->gc($maxlifetime);
```

Comment:
should  we remove the cast to let `false` get through?

Reply:
I've adjusted this to pass through the `false` return value. I would be happy to just drop it completely though, I'm not sure why these casts are present.

Reply:
I would say the cast was there to make implementations returning `int|false` satisfy the `@return bool` that was assumed. I don't think it makes sense to convert that to a `int` cast.